libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamOutput.java

```diff
@@ -804,6 +804,23 @@ private static Class<?> getGenericType(Object value) {
         }
     }
 
+    /**
+     * Returns the registered writer for the given class type.
+     */
+    @SuppressWarnings("unchecked")
+    public static <W extends Writer<?>> W getGenericValueWriter(Class<?> type) {
```

Comment:
Nice! Super simple win here. One little nitpick.
Let's change the name here to be more specific we're retrieving by class. Just a little easier to read scattered across the codebase.

```suggestion
    public static <W extends Writer<?>> W getValueWriterByClass(Class<?> type) {
```

Reply:
Sure! I've rename it to `getGenericValueWriterByClass`. Still keeping 'generic' in the name as the writer first writes the ordinal representing the type of the generic value, making it different from other kinds of non-generic writers.